apps/opik-backend/src/main/java/com/comet/opik/domain/ExperimentService.java

```diff
@@ -473,6 +473,35 @@ public Mono<Boolean> validateExperimentWorkspace(@NonNull String workspaceId, @N
                 .all(experimentWorkspace -> workspaceId.equals(experimentWorkspace.workspaceId()));
     }
 
+    @WithSpan
+    public Mono<Void> finishExperiments(@NonNull Set<UUID> ids) {
+        Preconditions.checkArgument(CollectionUtils.isNotEmpty(ids), "Argument 'ids' must not be empty");
+
+        return Mono.deferContextual(ctx -> {
+            String workspaceId = ctx.get(RequestContext.WORKSPACE_ID);
+            String workspaceName = ctx.get(RequestContext.WORKSPACE_NAME);
+            String userName = ctx.get(RequestContext.USER_NAME);
+
+            log.info("Finishing experiments, count '{}', workspaceId '{}'", ids.size(), workspaceId);
+
+            return experimentDAO.getByIds(ids)
+                    .collectList()
+                    .doOnNext(experiments -> {
+                        if (CollectionUtils.isNotEmpty(experiments)) {
+                            log.info("Raising alert event for finished experiments, count '{}'", experiments.size());
+                            eventBus.post(com.comet.opik.api.events.webhooks.AlertEvent.builder()
```

Comment:
Minor: Replace qualified name with import
```suggestion
                            eventBus.post(AlertEvent.builder()
```